lib/vmpooler/api/v1.rb

```diff
@@ -173,6 +173,72 @@ class V1 < Sinatra::Base
       JSON.pretty_generate(result)
     end
 
+    get "#{api_prefix}/token/:token/?" do
+      content_type :json
+
+      result = {}
+
+      Vmpooler::API.settings.config[:auth] ? status(401) : status(404)
+      result['ok'] = false
+
+      if Vmpooler::API.settings.config[:auth] and Vmpooler::API.settings.redis.exists('vmpooler__token__' + params[:token])
+        status(200)
+        result['ok'] = true
+
+        result[params[:token]] = Vmpooler::API.settings.redis.hgetall('vmpooler__token__' + params[:token])        
```

Comment:
If for some reason the key is deleted between the `exists` and the `hgetall`, Redis will return a `nil`.

````json
{
  "ok": true,
  "hwwzouwrg5a0k6ezk4lyy1ma4zzvtwg1": {

  }
}
````